src/components/QuranReader/index.tsx

```diff
@@ -132,22 +135,35 @@ const QuranReader = ({
   );
 };
 
+interface RequestKeyInput {
+  quranReaderDataType: QuranReaderDataType;
+  index: number;
+  initialData: VersesResponse;
+  quranReaderStyles: QuranReaderStyles;
+  selectedTranslations: number[];
+  selectedTafsirs: number[];
+  isVerseData: boolean;
+  isTafsirData: boolean;
+  id: string | number;
+  reciter: number;
+}
 /**
  * Generate the request key (the API url based on the params)
  * which will be used by useSwr to determine whether to call BE
  * again or return the cached response.
  */
-const getRequestKey = (
-  quranReaderDataType: QuranReaderDataType,
-  index: number,
-  initialData: VersesResponse,
-  quranReaderStyles: QuranReaderStyles,
-  selectedTranslations: number[],
-  selectedTafsirs: number[],
-  isVerseData: boolean,
-  isTafsirData: boolean,
-  id: string | number,
-): string => {
+const getRequestKey = ({
+  id,
+  isVerseData,
+  isTafsirData,
+  initialData,
+  index,
+  quranReaderStyles,
+  quranReaderDataType,
+  selectedTranslations,
+  selectedTafsirs,
+  reciter,
+}: RequestKeyInput): string => {
   // if the response has only 1 verse it means we should set the page to that verse this will be combined with perPage which will be set to only 1.
   const page = isVerseData || isTafsirData ? initialData.verses[0].verseNumber : index + 1;
   if (quranReaderDataType === QuranReaderDataType.Juz) {
```

Comment:
The `reciter` should be passed here too because the user can be reading a Juz in the translation mode and he might want to play a verse (that's to assume the Juz's API support that).